batch/batch/batch.py

```diff
@@ -189,6 +193,18 @@ class Job:
                 container.volume_mounts = []
             container.volume_mounts.extend(volume_mounts)
 
+        if self._current_task.name == 'main':
+            success_file = f'/io/__BATCH_{self._current_task.name.upper()}_SUCCESS__'
+            sh_expression = f'(test -e {success_file} && rm {success_file} && exit 1) || ' \
```

Comment:
if the `rm` fails for some reason this will touch and exit 0. I think you want a semicolon to join the rm and the exit.